### DISCUSSION TITLE
Five server runs class ``constructor`` twice, making Web Components development *impossible*. Switching back to Live Server.

### DISCUSSION THREAD

**Discussion by Danny-Engelman** [yandeu/five-server#34] [Help / Questions] [answered]
With this code, **Five Server** seems to run the ``constructor`` **twice**

````html
<!DOCTYPE html>
<html lang="en">
  <head>
  </head>
  <body>
    <main-component id="ONE">Hello Web Component</main-component>
    <script>
        customElements.define(
          "main-component",
          class extends HTMLElement {
            constructor() {
              super();
              console.log(`constructor <${this.nodeName}>`, this.id);
            }
          } );
      </script>
    </body>
</html>
````

![](https://i.imgur.com/Qdw8Vzb.png)

I have disabled all Extensions.

Running inJSFiddle, CodePen or from localfile  
**or** with **Live** Server the ``constructor`` is only executed **once**

I switched from Live to Five because of SVG issues, switching back again...

I presume Five Server is doing some ``createElement(..)`` checking or DOM parsing under the hood which triggers the ``constructor``
I get an extra ``constructor`` call for every DOM Custom Element

**Comment by yandeu** [comment, marked as answer]
Try:

```html
<!DOCTYPE html>
<html>
  <body>
    <main-component id="ONE">Hello Web Component</main-component>
    <script>
      const main = () => {
        customElements.define(
          'main-component',
          class extends HTMLElement {
            constructor() {
              super()
              console.log(`constructor <${this.nodeName}>`, this.id)
            }
          }
        )
      }

      // wait for five-server to connect
      const five = document.querySelector('[data-id="five-server"]')
      if (five) five.addEventListener('connected', main)
      else window.addEventListener('load', main)
    </script>
  </body>
</html>

```

**Reply by Danny-Engelman** [reply]
Sorry, I can't add code just because a Tool doesn't play right.
Besided Web Components can be loaded in the ``<head>``; not long after in the ``load`` Event

**Reply by yandeu** [reply]
Then please open a [new issue](https://github.com/yandeu/five-server/issues/new/choose) and it will be fixed.

**Reply by yandeu** [reply]
I have fixed it! Will work in the next update as long as you don't use the `injectBody` feature.

See https://github.com/yandeu/five-server/commit/d6d4230324d243d12ea9949fff5fe5482ffdb5a9
